kpi/views/v2/data.py

```diff
@@ -365,6 +370,18 @@ def duplicate(self, request, pk, *args, **kwargs):
         """
         Creates a duplicate of the submission with a given `pk`
         """
+        # User needs both `change_submissions` and `add_submissions`
+        # permissions in order to create a duplicate
+        if not all(
+            perm in self.asset.get_perms(request.user)
+            for perm in [PERM_ADD_SUBMISSIONS, PERM_CHANGE_SUBMISSIONS]
+        ):
+            raise exceptions.PermissionDenied(
+                detail=_(
+                    'User needs both `add_submissions` and '
+                    '`change_submissions` permissions to duplicate a submission.'
+                )
+            )
```

Comment:
Not needed, `DuplicateSubmissionPermission` should do the job when the correct permissions are set in `perms_map`